app/gui2/src/stores/graph/imports.ts

```diff
@@ -341,6 +291,47 @@ export function filterOutRedundantImports(
   return required.filter((info) => !existing.some((existing) => covers(existing, info)))
 }
 
+/* Information about detected conflict import, and advisory on resolution. */
+export interface DetectedConflict {
+  /* Always true, for more expressive API usage. */
+  detected: boolean
+  /* Advisory to replace the following name (qualified name or single ident)… */
+  pattern: QualifiedName | IdentifierOrOperatorIdentifier
+  /* … with this fully qualified name. */
+  fullyQualified: QualifiedName
+}
+
+export type ConflictInfo = DetectedConflict | undefined
+
+/* Detect possible name clash when adding `importsForEntry` with `existingImports` present. */
+export function detectImportConflicts(
+  suggestionDb: SuggestionDb,
+  existingImports: Import[],
+  importToCheck: RequiredImport,
+): ConflictInfo {
+  const entryFQN = entryFQNFromRequiredImport(importToCheck)
+  const [entryId] = suggestionDb.nameToId.lookup(entryFQN)
+  if (entryId == null) return
+  const entry = suggestionDb.get(entryId)!
+  const conflictingIds = suggestionDb.conflictingNames.lookup(entry.name)
```

Comment:
Technically, we could just read the last segment of `entryFQN`, as name, and don't fetch the entry (only `entryId`).

Reply:
True, fixed